srcs/app/server/WebServer.cpp

```diff
@@ -35,72 +36,23 @@ void	WebServer::Run() {
 	}
 }
 
-int		WebServer::BindNewListeningSocketToServer_(const ServerConfig &settings) {
-	int listen_sd = socket(AF_INET, SOCK_STREAM, 0);
-	if (listen_sd < 0) {
-		throw std::runtime_error(std::strerror(errno));
-	}
-	if (fcntl(listen_sd, F_SETFL, O_NONBLOCK) < 0) {
-		throw std::runtime_error(std::strerror(errno));
-	}
-
-	struct sockaddr_in addr;
-	addr.sin_family = AF_INET;  // IPv4
-	addr.sin_port = htons(settings.listen_port);
-	addr.sin_addr.s_addr = htonl(settings.listen_address);
-	std::memset(addr.sin_zero, 0, sizeof(addr.sin_zero));
-
-	int on = 1;
-	if (setsockopt(listen_sd, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on)) < 0) {
-		throw std::runtime_error(std::strerror(errno));
-	}
-
-	if (bind(listen_sd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
-		throw std::runtime_error(std::strerror(errno));
-	}
-
-	if (listen(listen_sd, SOMAXCONN) < 0) {
-		throw std::runtime_error(std::strerror(errno));
-	}
-	return listen_sd;
-}
-
 void	WebServer::PopulateServers_() {
 	std::vector<ServerConfig>	servers_settings;
 
 	servers_settings = config_.GetServersSettings();
 	std::vector<ServerConfig>::iterator	settings_it = servers_settings.begin();
 
 	while (settings_it != servers_settings.end()) {
-		int listen_sd = BindNewListeningSocketToServer_(*settings_it);
+		int listen_sd = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
TODO: Error checking

Reply:
What do you guys think about the SyscallWrapper class of static methods that do syscalls and throw in case of error?

Reply:
Just in case, there is a if guard in `Server` class for this case, but will move it to this class since makes more sense. Thanks!